### ISSUE TITLE
[React Native] Can currently only create a blob from other Blobs

### ISSUE THREAD

**Issue by hasancanozkan** [austinksmith/Hamsters.js#40]
Hi, I have a RN application and just want to try parallel programming using hamsters.js. First, I created a demo app to try it with a worker, namely https://github.com/fabriciovergal/react-native-workers.  however I am getting a strange error,
What version of the library --> 4.2.0
What operating system are you seeing the issue on? --> Windows
Which platform (and or) browser are you seeing the issue on? React Native on andorid
What is the version of the platform (and or) browser you are seeing the issue on? --> RN 48.0
The code that I wrote simply (i saw this in how the library manages your data) : 
initHamsters(){
        const startOptions = {
            maxThreads: 4,
            cache: false,
            debug: false,
            persistence: false,
            Worker: Worker
        };
        hamsters.init(startOptions);
        const params = {
            array: [1,2,3,4,5,6,7,8,9,10],
            animal: 'Hamster',
        };
        hamsters.run(params, function() {
            if(params.animal === 'Hamster') {
                rtn.data.push('Hamsters are awesome');
            }
        }, function(result) {
            console.log(result);
        });
    }
I have to say that there is not so many worker libraries availalbler for react native, and the one that I found was working, i tried it. However, it has some requirements like using two ports and having index.worker.js  and I didnt provide them to run this code. the error is like this : 
![img_1313](https://user-images.githubusercontent.com/15143230/30073766-65465e88-926f-11e7-8aae-4aa135db3672.JPG)
and as I understand this is a react native error, which doesnt allow to use this library and Worker library. 
Any ideas?

**Comment by austinksmith** [comment]
This is an interesting issue and it seems to revolve around ReactNative not
supporting data blobs all of a sudden? In the past data blobs worked fine
and unfortunately the library is somewhat reliant on that capability since
its the most efficient way to accomplish the functionality I needed, I will
have to look into this deeper and possibly even change how data is being
passed in the event it's running inside ReactNative. If you know of a
external module to make use of DataBlobs within ReactNative please give
that a try and let me know if that helps at all. If it's necessary now to
use a third party DataBlob plugin then that is unfortunate but at least I
can ensure the documentation reflects that.

This message (including any attachments) may contain confidential,
proprietary, privileged and/or private information. The information is
intended to be for the use of the individual or entity designated above. If
you are not the intended recipient of this message, please notify the
sender immediately, and delete the message and any attachments. Any
disclosure, reproduction, distribution or other use of this message or any
attachments by an individual or entity other than the intended recipient is
prohibited.

**Comment by austinksmith** [comment]
Cross referencing this bug report https://github.com/facebook/react-native/issues/16034

**Comment by amos80m** [comment]
same problem

**Comment by austinksmith** [comment]
We are still waiting for this to land, looks like it wont be official until 0.53 at the earliest. https://github.com/facebook/react-native/pull/11573 

Fortunately the new version of Hamsters.js will include a workaround for this, though it's not ideal it should get us past this issue once and for all.

**Comment by satya164** [comment]
> This is an interesting issue and it seems to revolve around ReactNative not supporting data blobs all of a sudden

FYI React Native never supported Blobs.

**Comment by amos80m** [comment]
I moved my code to Heroku => post the image file and send it to Firestore.
working 100%

**Comment by austinksmith** [comment]
@satya164  I can definitively say the library worked flawlessly (Android 4.0+) when I first started developing using ReactNative about a year and a half ago. Blob support was never an issue, whether it really worked or just failed silently the library never had an issue running until recent version of RN were released. Perhaps it has something to do with their push for the new fiber stuff. 

There is a solution to this problem but it requires using the external `common/wheel.min.js` file instead of blobs similarly to how IE10 support works (microsoft security problems)

**Comment by austinksmith** [comment]
@amos80m You solved the problem within Hamsters.js or the problem within ReactNative? I'm not sure i follow.

**Comment by satya164** [comment]
It's not related to Fiber. There was no `Blob` object available earlier. Blob support was added recently, but it's not full-featured yet. If your code had a fallback without blobs, it was probably working, and if not, it was failing silently.

**Comment by austinksmith** [comment]
Can you give dev branch a try alongside react-native-threads? Just make sure you import the library as laid out below. Apologies for the long delay, next version is a complete ground up rewrite of the library and its taken a long time to get it where I want. 

```
import { Thread } from 'react-native-threads';
```

and then do this on your library startup

```
hamsters.init({
  Worker: Thread
});
```

Please report back your results, I'm nearing completion on the next library version and react native support is important, thanks!

**Comment by austinksmith** [comment]
I do not have access to any systems that will allow me to test react native support especially for IOS or AppleTV etc. I will need the community to validate this works.

**Comment by amos80m** [comment]
@austinksmith I created a Node.j.s App that receive images files as post from react native app.
the Node.js App is connected as an Admin to firebase and saving my uploaded images to firestore.
and i can upload as many images and sizes as I need without any issues

**Comment by austinksmith** [comment]
I've released version 5.0.0 of Hamsters.js, this should resolve any outstanding react native issues but please open a ticket if it does not :+1:  thanks!

https://github.com/austinksmith/Hamsters.js/releases/tag/v5.0.0
